moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp

```diff
@@ -1422,8 +1422,9 @@ void MotionPlanningDisplay::fixedFrameChanged()
   if (int_marker_display_)
     int_marker_display_->setFixedFrame(fixed_frame_);
   // When the fixed frame changes we need to tell RViz to update the rendered interactive marker display
-  frame_->scene_marker_->requestPoseUpdate(frame_->scene_marker_->getPosition(),
-                                           frame_->scene_marker_->getOrientation());
+  if (frame_ && frame_->scene_marker_)
+    frame_->scene_marker_->requestPoseUpdate(frame_->scene_marker_->getPosition(),
+                                             frame_->scene_marker_->getOrientation());
```

Comment:
Should we log something when this is false? Ideally, we'd send the error some other way... I generally don't like reporting errors in logs if we can help it but it is definitely better to not dereference an invalid pointer than what is here.

Reply:
I took a deeper look, and `frame_` is valid but `frame->scene_marker_` is not.

This is totally fine given how `MotionPlanningFrame::createSceneInteractiveMarker()` in `moveit2/moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_objects.cpp` doesn't always instantiate a scene marker -- in fact, this code path only makes a marker if we have a collision object selected in the scene, as in here:

![image](https://user-images.githubusercontent.com/4603398/192381958-38f786f2-ffc6-484a-b495-be5ebbbfeefa.png)

Also, only a few lines above we have a similar valid pointer check for the `int_marker_display_` variable (which is the interactive marker on the move group's end effector frame) and we don't log anything on that pointer being null.

Reply:
Thank you for the detailed response.